toucan_connectors/redshift/redshift_database_connector.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+import redshift_connector
+from pydantic import Field, SecretStr
+from pydantic.types import constr
+from redshift_connector.error import InterfaceError
+
+from toucan_connectors.common import ConnectorStatus
+from toucan_connectors.toucan_connector import ToucanConnector, ToucanDataSource
+
+
+class RedshiftDataSource(ToucanDataSource):
+    table: str = Field(None, description='The name of the data table that you want to ')
+    query: constr(min_length=1) = Field(
+        description='A string describing a query (CAUTION: Use limit to avoid to retrieve too many datas)',
+        widget='sql',
+    )
+
+
+class RedshiftConnector(ToucanConnector):
+    data_source_model: RedshiftDataSource
+
+    user: str = Field(..., description='Your login username.')
+    password: SecretStr = Field(None, description='Your login password')
+    host: str = Field(None, description='IP address or hostname.')
+    port: int = Field(..., description='The listening port of your Redshift Database')
+    database: str = Field(..., description='The name of the database you want to query')
```

Comment:
database should be moved to the datasource, thus we can configure it when editing a query.